File: app/src/main/java/com.example.safehome/Door/Door.java
package com.example.safehome.Door;

import android.icu.lang.UCharacter;

public interface Door <Object> {

    /**
     * detects whether an event has occurred
     */
    DType determineType();

    /**
     * identifies the type of sensor
     */
    void identify();


    /**
     * enables notifications
     */
    void enableNotification();

    /**
     * disables notifications
     */
    void disableNotification();

    /**
     * reconfigures sensor
     */
    void reconfigure();

    /**
     * adds sensor to system
     */
    void addSensorMomento();

    /**
     * gets existing sensor from system
     */
    void getSensorMomento();
}

File: app/src/main/kotlin/com.example.safehome/Door/Door.kt
package com.example.safehome.Door


interface Door {
    /**
     * detects whether an event has occurred
     */
    fun determineType(): DType?

    /**
     * identifies the type of sensor
     */
    fun identify()

    /**
     * enables notifications
     */
    fun enableNotification()

    /**
     * disables notifications
     */
    fun disableNotification()

    /**
     * reconfigures sensor
     */
    fun reconfigure()

    /**
     * adds sensor to system
     */
    fun addSensorMomento()

    /**
     * gets existing sensor from system
     */
    fun getSensorMomento()
}